numba_typing/local_variable_type_annotations.py

```diff
@@ -29,27 +30,33 @@ def visit_AnnAssign(self, node):
         if isinstance(annotation, ast.Subscript):  # containers and generics
             try:
                 container_name = annotation.value.id
-                if isinstance(annotation.slice.value, ast.Tuple):
-                    types_as_str = ','.join(elt.id for elt in annotation.slice.value.elts)
-                    exec_variables = f'{target.id} = {self.module_name}.{container_name}[{types_as_str}]'
-                else:
-                    exec_variables = f'{target.id} = {self.module_name}.{container_name}[{annotation.slice.value.id}]'
+                module_import_name = ''
             except AttributeError:  # typing.
-                module_import_name = annotation.value.value.id
+                module_import_name = annotation.value.value.id + '.'
```

Comment:
We could and should avoid this approach. We can extract all global and non-local variables from function object itself instead of extracting objects from module.

Function object contains list of globals it captured in `__globals__` member.
Also you should use `inspect.getclosurevars` to get captured non-locals for the following case:
```python
def foo():
    T = TypeVar('T')
    def bar():
        a: T
    ...

    return bar


my_func = foo()
```

So, instead of trying to get objects from module, we should use function global and closure vars, unite it in single dictionary and pass it to `exec` as globals.